gen/plugin_test.go

```diff
@@ -9,13 +9,33 @@ import (
 	"go.uber.org/thriftrw/ptr"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 const (
 	_testPackagePrefix = "go.uber.org/thriftrw/gen/internal/tests"
 	_testThriftRoot    = "idl"
 )
 
+func TestAddRootModule(t *testing.T) {
+	spec := &compile.ServiceSpec{
+		Name: "EmptyService",
+		File: "idl/empty.thrift",
+	}
+
+	importer := thriftPackageImporter{
+		ImportPrefix: "go.uber.org/thriftrw/gen/internal/tests",
+		ThriftRoot:   "idl",
+	}
+
+	g := newGenerateServiceBuilder(importer)
+
+	_, err := g.AddRootModule(spec.ThriftFile())
+	require.NoError(t, err)
+
+	assert.Len(t, g.rootModules, 1)
+}
+
 func TestAddRootService(t *testing.T) {
```

Comment:
should we rename this test?

Reply:
Renamed to include RootModulesandServices to reflect additional functionality.